core/tests/test_views.py

```diff
@@ -72,39 +94,66 @@ def test_add_post_unauthorized(client: Client, community: Community) -> None:
-    post.refresh_from_db()
-    assert response.context["comments"][0].author == user
-    assert response.context["comments"][0].content == data["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
+        image = Image.new("RGB", (100, 100), color=(73, 109, 137))
+        image.save(tmp_file, format="JPEG")
+        tmp_file.seek(0)
+
+        avatar_file = SimpleUploadedFile(name="test_avatar.jpg", content=tmp_file.read(), content_type="image/jpeg")
+
+        user.avatar = avatar_file
+        user.save()
+        data = {
+            "parent_id": post.pk,
+            "content": "This is a test comment content.",
+        }
+        client.force_login(user)
+        assert post.children_count == 0
+        assert post.get_comments().count() == 0
+        response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
+        assert response.status_code == 200
+        assert post.children_count == 1
+        assert post.get_comments().count() == 1
+        post.refresh_from_db()
+        assert response.context["comments"][0].author == user
+        assert response.context["comments"][0].content == data["content"]
 
 
 def test_add_comment_invalid(client: Client, user: User, post: Post) -> None:
-    data = {"parent_id": post.pk, "content": ""}
-    client.force_login(user)
-    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
-    assert response.status_code == 200
-    form = response.context["form"]
-    assert len(form.errors) == 1
-    assert "This field is required." in form.errors["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
+        image = Image.new("RGB", (100, 100), color=(73, 109, 137))
+        image.save(tmp_file, format="JPEG")
+        tmp_file.seek(0)
+
+        avatar_file = SimpleUploadedFile(name="test_avatar.jpg", content=tmp_file.read(), content_type="image/jpeg")
+
+        user.avatar = avatar_file
+        user.save()
+        data = {"parent_id": post.pk, "content": ""}
+        client.force_login(user)
+        response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
+        assert response.status_code == 200
+        form = response.context["form"]
+        assert len(form.errors) == 1
+        assert "This field is required." in form.errors["content"]
 
 
 def test_add_comment_valid_special_characters(client: Client, user: User, post: Post) -> None:
-    data = {"parent_id": post.pk, "content": "This is a test comment with special characters! 😊🚀✨"}
-    client.force_login(user)
-    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
-    assert response.status_code == 200
-    new_comment = Post.objects.get(content=data["content"])
-    assert new_comment.content == data["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
```

Comment:
### Test refactoring needed

Duplicated code. I am thinking we need only one place to see if the avatar is shown. 
Actually, what I would to is create couple simple tests
- create new file with the tests described below
- a post with data to see if the user avatar is displayed on the post (if the URL is ok), when user has avatar
- as above for the comment
- repeat the above tests for default avatar image. 
- for all the above tests I would simply create commends using query `Post.object.create...` instead of HTTP requests - the goal is to see user avatar on post displayed, not to verify if the form to add post works. 

Thats it! Do not modify any of the existing tests. This only adds more complication, we only need to see if avatar is displayed.  

The context manager usage is great here, but we really don't need more complicated logic in tests. It slows them down, and tests need to be fast. It also adds more complexity and burden in future maintenance. Let's keep it simple. 

Please refactor.

Reply:
Okay, I'll do it, I just need time for these tests